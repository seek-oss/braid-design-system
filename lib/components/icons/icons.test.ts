import globby from 'globby';
import path from 'path';
import fs from 'fs';

const svgComponentPaths = globby.sync('*Icon/*Svg.tsx', {
  cwd: __dirname,
  absolute: true,
});

svgComponentPaths.forEach(svgComponentPath => {
  const componentName = path.basename(svgComponentPath, '.tsx');

  describe(componentName, () => {
    it('should match snapshot', () => {
      const source = fs.readFileSync(svgComponentPath, 'utf-8');
      expect(source).toMatchSnapshot();
    });
  });
});
