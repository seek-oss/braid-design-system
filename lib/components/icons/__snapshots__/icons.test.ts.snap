// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBookmarkActiveSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconBookmarkActiveSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path
      d=\\"M18 21l-6-4-6 4V5c-.1-1 .7-1.9 1.7-2h8.6c1 .1 1.8 1 1.7 2v16z\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
    />
  </svg>
);
"
`;

exports[`IconBookmarkSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconBookmarkSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path
      d=\\"M18 21l-6-4-6 4V5c-.1-1 .7-1.9 1.7-2h8.6c1 .1 1.8 1 1.7 2v16z\\"
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
    />
  </svg>
);
"
`;

exports[`IconChevronSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconChevronSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
      d=\\"M20 8l-8 8-8-8\\"
    />
  </svg>
);
"
`;

exports[`IconClearSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconClearSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
      d=\\"M18 6L6 18M6 6l12 12\\"
    />
  </svg>
);
"
`;

exports[`IconCriticalSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconCriticalSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <circle
      cx={12}
      cy={12}
      r={10}
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeMiterlimit={10}
    />
    <circle cx={12} cy={17} r={1} />
    <path
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
      d=\\"M12 8v5\\"
    />
  </svg>
);
"
`;

exports[`IconDateSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconDateSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path d=\\"M19 4h-2V3c0-.6-.4-1-1-1s-1 .4-1 1v1H9V3c0-.6-.4-1-1-1s-1 .4-1 1v1H5C3.3 4 2 5.3 2 7v12c0 1.7 1.3 3 3 3h14c1.7 0 3-1.3 3-3V7c0-1.7-1.3-3-3-3zM4 7c0-.6.4-1 1-1h2v1c0 .6.4 1 1 1s1-.4 1-1V6h6v1c0 .6.4 1 1 1s1-.4 1-1V6h2c.6 0 1 .4 1 1v3H4V7zm16 12c0 .6-.4 1-1 1H5c-.6 0-1-.4-1-1v-7h16v7z\\" />
  </svg>
);
"
`;

exports[`IconInfoSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconInfoSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <circle
      cx={12}
      cy={12}
      r={10}
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
    />
    <path
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
      d=\\"M12 16v-4m0-4h0\\"
    />
  </svg>
);
"
`;

exports[`IconLocationSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconLocationSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path d=\\"M12 1C7.6 1 4 4.6 4 9c0 4.1 6.5 12.6 7.2 13.6.2.2.5.4.8.4s.6-.1.8-.4c.7-1 7.2-9.5 7.2-13.6 0-4.4-3.6-8-8-8zm0 19.3c-2.2-3-6-8.8-6-11.3 0-3.3 2.7-6 6-6s6 2.7 6 6c0 2.5-3.8 8.3-6 11.3z\\" />
    <path d=\\"M12 5c-1.7 0-3 1.3-3 3s1.3 3 3 3 3-1.3 3-3-1.3-3-3-3zm0 4c-.6 0-1-.4-1-1s.4-1 1-1 1 .4 1 1-.4 1-1 1z\\" />
  </svg>
);
"
`;

exports[`IconMoneySvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconMoneySvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path d=\\"M17 15.5c0-1.6-1.3-4.3-4.9-4.5-3-.2-3.1-2.5-3.1-2.5 0-.7.3-1.3.8-1.7.6-.5 1.4-.8 2.2-.8.8 0 1.6.3 2.2.8.4.4 1.1.3 1.4-.1.4-.4.3-1.1-.1-1.4-.7-.6-1.6-1-2.5-1.2V3c0-.6-.4-1-1-1s-1 .4-1 1v1.1c-.9.2-1.7.5-2.5 1.2C7.6 6.1 7 7.2 7 8.5c0 1.6 1.3 4.3 4.9 4.5 3 .2 3.1 2.5 3.1 2.5 0 .7-.3 1.3-.8 1.7-.6.5-1.4.8-2.2.8-.8 0-1.6-.3-2.2-.8-.4-.4-1.1-.3-1.4.1-.4.4-.3 1.1.1 1.4.7.6 1.6 1 2.5 1.2V21c0 .6.4 1 1 1s1-.4 1-1v-1.1c.9-.2 1.7-.5 2.5-1.2.9-.8 1.5-1.9 1.5-3.2z\\" />
  </svg>
);
"
`;

exports[`IconPositiveSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconPositiveSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <circle
      cx={12}
      cy={12}
      r={10}
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeMiterlimit={10}
    />
    <path
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
      d=\\"M16 10l-5 5-2-2\\"
    />
  </svg>
);
"
`;

exports[`IconSearchSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconSearchSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path d=\\"M21.7 20.3L18 16.6c1.2-1.5 2-3.5 2-5.6 0-5-4-9-9-9s-9 4-9 9 4 9 9 9c2.1 0 4.1-.7 5.6-2l3.7 3.7c.2.2.5.3.7.3s.5-.1.7-.3c.4-.4.4-1 0-1.4zM4 11c0-3.9 3.1-7 7-7s7 3.1 7 7c0 1.9-.8 3.7-2 4.9-1.3 1.3-3 2-4.9 2-4 .1-7.1-3-7.1-6.9z\\" />
  </svg>
);
"
`;

exports[`IconSocialFacebookSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconSocialFacebookSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path d=\\"M13.2 20v-7.3h2.5l.4-2.9h-2.8V8c0-.8.2-1.4 1.4-1.4h1.5V4.1C15.4 4 14.7 4 14 4c-1.9-.1-3.5 1.3-3.7 3.2v2.7H7.8v2.9h2.5V20h2.9z\\" />
  </svg>
);
"
`;

exports[`IconSocialTwitterSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconSocialTwitterSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path d=\\"M9 18.5c5.1 0 9.3-4.1 9.3-9.2v-.5C19 8.3 19.6 7.7 20 7c-.6.3-1.2.4-1.9.5.7-.4 1.2-1.1 1.4-1.8-.6.4-1.3.7-2.1.8-1.2-1.3-3.3-1.4-4.6-.1-.9.8-1.2 2-.9 3.1-2.7-.1-5.1-1.3-6.8-3.4-.9 1.5-.4 3.4 1 4.4-.5 0-1-.2-1.5-.4 0 1.6 1.1 2.9 2.6 3.2-.5.1-1 .2-1.5.1.4 1.3 1.7 2.3 3.1 2.3-1.2.9-2.6 1.4-4.1 1.4h-.8c1.6.9 3.3 1.4 5.1 1.4\\" />
  </svg>
);
"
`;

exports[`IconTickSvg should match snapshot 1`] = `
"import React, { AllHTMLAttributes } from 'react';

export const IconTickSvg = (props: AllHTMLAttributes<SVGElement>) => (
  <svg
    viewBox=\\"0 0 24 24\\"
    focusable=\\"false\\"
    fill=\\"currentColor\\"
    width={16}
    height={16}
    {...props}
  >
    <path
      fill=\\"none\\"
      stroke=\\"currentColor\\"
      strokeWidth={2}
      strokeLinecap=\\"round\\"
      strokeLinejoin=\\"round\\"
      d=\\"M19 7L9 17l-4-4\\"
    />
  </svg>
);
"
`;
